pylot/prediction/r2p2_predictor_operator.py

```diff
@@ -38,150 +37,253 @@ class R2P2PredictorOperator(erdos.Operator):
         tracking_stream.add_callback(self.on_trajectory_update)
         erdos.add_watermark_callback(
-            [pose_stream, point_cloud_stream, tracking_stream],
+            [point_cloud_stream, tracking_stream],
             [prediction_stream], self.on_watermark)
 
-        self._vehicle_id_stream = vehicle_id_stream
         self._lidar_setup = lidar_setup
 
-        self._can_bus_msgs = deque()
         self._point_cloud_msgs = deque()
         self._tracking_msgs = deque()
 
         
     @staticmethod
-    def connect(can_bus_stream, point_cloud_stream, tracking_stream, vehicle_id_stream):
+    def connect(point_cloud_stream, tracking_stream):
         prediction_stream = erdos.WriteStream()
         return [prediction_stream]
 
 
-    def run(self):
-        # Read the ego vehicle id from the vehicle id stream.
-        vehicle_id_msg = self._vehicle_id_stream.read()
-        self.vehicle_id = vehicle_id_msg.data
-        self._logger.debug(
-            "The R2P2PredictorOperator received the vehicle id: {}".format(
-                self.vehicle_id))
-
-
     def on_watermark(self, timestamp, prediction_stream):
         self._logger.debug('@{}: received watermark'.format(timestamp))
 
-        can_bus_msg = self._can_bus_msgs.popleft()
         point_cloud_msg = self._point_cloud_msgs.popleft()
         tracking_msg = self._tracking_msgs.popleft()
 
-        binned_lidar = self.get_occupancy_grid(point_cloud_msg.point_cloud.points)
-
-        self._vehicle_transform = can_bus_msg.data.transform
-        
-        # Get the ego-vehicle track and bounding box.
-        ego_vehicle_trajectory = None
-        for vehicle in tracking_msg.obstacle_trajectories:
-            if vehicle.id == self.vehicle_id:
-                ego_vehicle_trajectory = vehicle.trajectory
-                ego_vehicle_bbox = vehicle.bounding_box
-                break
-        assert ego_vehicle_trajectory is not None, "ego-vehicle trajectory not found"
-        ego_vehicle_trajectory = np.stack(
-            [[point.location.x, point.location.y] for
-              point in ego_vehicle_trajectory])
-
-        # If we haven't seen enough past locations yet, pad the start
-        # of the trajectory with copies of the earliest location.
-        num_past_locations = ego_vehicle_trajectory.shape[0]
-        if num_past_locations < self._flags.prediction_num_past_steps:
-            initial_copies = np.repeat([np.array(ego_vehicle_trajectory[0])],
-                                       self._flags.prediction_num_past_steps - num_past_locations,
-                                       axis=0)
-            ego_vehicle_trajectory = np.vstack(
-                (initial_copies, ego_vehicle_trajectory))
-        ego_vehicle_trajectory = np.expand_dims(ego_vehicle_trajectory, axis=0)
-
+        all_vehicles = [trajectory for trajectory in tracking_msg.obstacle_trajectories if trajectory.label == 'vehicle']
```

Comment:
Can you please check if label is in pylot.perception.detection.utils.VEHICLE_LABELS? This makes sure that the code works even if we use real perception.